src/api/composition/DefaultCompositionProvider.js

```diff
@@ -68,45 +65,78 @@ define([
     /**
      * Load any domain objects contained in the composition of this domain
      * object.
-     * @param {module:openmct.DomainObjcet} domainObject the domain object
+     * @param {module:openmct.DomainObject} domainObject the domain object
      *        for which to load composition
-     * @returns {Promise.<Array.<module:openmct.DomainObject>>} a promise for
-     *          the domain objects in this composition
+     * @returns {Promise.<Array.<module:openmct.Identifier>>} a promise for
+     *          the Identifiers in this composition
      * @memberof module:openmct.CompositionProvider#
      * @method load
      */
     DefaultCompositionProvider.prototype.load = function (domainObject) {
-        return Promise.all(domainObject.composition.map(ObjectAPI.get));
+        return Promise.all(domainObject.composition);
     };
 
+    /**
+     * Attach listeners for changes to the composition of a given domain object.
+     * Supports `add` and `remove` events.
+     *
+     * @param {module:openmct.DomainObject} domainObject to listen to
+     * @param String event the event to bind to, either `add` or `remove`.
+     * @param Function callback callback to invoke when event is triggered.
+     * @param [context] context to use when invoking callback.
+     */
     DefaultCompositionProvider.prototype.on = function (
         domainObject,
         event,
-        listener,
+        callback,
         context
```

Comment:
Ah, yeah, un-listening was the main reason I could think of for doing it this way, but I figured we are mainly using bound member functions where a reference is retained anyway. I see the value in allowing flexibility here though, so happy to leave it as is.